### PR TITLE
Checkout

### LINKED ISSUE #29: API to checkout a commit or reference

### PR DESCRIPTION
**Pull request by mdiep**
Fixes #29.

Remaining tasks:
- [x] Add strategy parameter
- [x] Add progress callbacks
- [x] ~~Test `Repository.checkout()`?~~

### CODE REVIEW

**Comment by mdiep** [comment]
This is ready for review.

File: SwiftGit2/SwiftGit2.h [hunk 1]
```diff
@@ -17,3 +17,9 @@ FOUNDATION_EXPORT const unsigned char SwiftGit2VersionString[];
 // In this header, you should import all the public headers of your framework using statements like #import <SwiftGit2/PublicHeader.h>
 
 #import "git2.h"
+
+typedef void (^SG2CheckoutProgressBlock)(NSString * __nullable, NSUInteger, NSUInteger);
+
+/// A C function for working with Libgit2. This shouldn't be called directly. It's an
+/// implementation detail that, unfortunately, leaks through to the public headers.
+extern git_checkout_options SG2CheckoutOptions(SG2CheckoutProgressBlock __nullable progress);
```
Comment (modocache) [inline SwiftGit2/SwiftGit2.h]: Ah, too bad this hasn't been fixed yet. Nice workaround!

File: SwiftGit2Tests/RepositorySpec.swift [hunk 2]
```diff
@@ -429,5 +429,71 @@ class RepositorySpec: QuickSpec {
...
+				expect(repo.setHEAD(branch)).to(haveSucceeded())
+				expect(repo.HEAD().value?.shortName).to(equal(branch.name))
+				
+				expect(repo.setHEAD(oid)).to(haveSucceeded())
+				expect(repo.HEAD().value?.longName).to(equal("HEAD"))
+			}
+		}
+		
+		describe("Repository.checkout()") {
+			// We're not really equipped to test this yet. :(
```
Comment (modocache) [inline SwiftGit2Tests/RepositorySpec.swift]: nitpick: Perhaps use `xdescribe` with `it` closures explaining what you would test, if you could?

**Comment by modocache** [comment]
I had a suggestion for the test case, but other than that looks great! :+1:

**Pull request merged**